### DISCUSSION TITLE
docker run from Dockerfile

### DISCUSSION THREAD

**Discussion by mediumhust** [TinyHust/notes#207] [DevOps]
```
docker build --target run -t tinyhust/od-api:latest . |  grep "Successfully built" | sed 's/Successfully built //g' | xargs -I{} docker run {}
```

**Comment by SelinaCrocketta** [comment]
This command performs the following tasks:

Builds a Docker image using the Dockerfile in the current directory, with the build target set to "run", and tags the resulting image as "tinyhust/od-api:latest".
The output of the "docker build" command is piped to "grep" to filter out any extraneous information and only show the line that contains "Successfully built".
The output from "grep" is piped to "sed" to remove the "Successfully built" prefix from the image ID.
Finally, "xargs" is used to pass the image ID as an argument to "docker run", which starts a container based on the newly built image.
In short, this command builds a Docker image and runs a container using that image.
